File: src/main/scala/com/celestia/MainFrame.scala

package com.celestia

import java.awt.Dimension
import java.awt.BorderLayout 
import javax.swing.JFrame

import com.jogamp.opengl.util.FPSAnimator

//import com.jogamp.opengl.GLEventListener
//import com.jogamp.opengl.GLAutoDrawable
import com.jogamp.opengl.awt.GLCanvas

import com.celestia.util.R
import com.celestia.util.GLEventHandler


/**
  * Main Frame class
  * 
  * This class is the primary Frame for handling GL drawing.
  * It contains a GLEventListener implementation for handling
  * user input, and a GLCanvas object for drawing to the 
  * graphics card.
  */
class MainFrame extends JFrame {

  // General variables to do with JFrame dimensions 
  lazy val aspectRatio: Double = 9.0 / 16.0 
  lazy val defaultWidth: Int = 1000
  lazy val defaultHeight: Int = (defaultWidth * aspectRatio).toInt 
  lazy val Size: Dimension = new Dimension(defaultWidth, defaultHeight)

  // GLCanvas object for rendering
  lazy val glEVentHandler: GLEventHandler = new GLEventHandler
  lazy val glCanvas: GLCanvas = new GLCanvas
  lazy val fpsAnimator:FPSAnimator = new FPSAnimator(glCanvas, R.util.fps)

  /**
   * Initializer for creating the base JFrame object
   */
  def setup {
    setTitle(R.util.title)
    setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE)
    setLayout(new BorderLayout)
    getContentPane.add(glCanvas)
    glCanvas.addGLEventListener(glEVentHandler)
    glCanvas.setAnimator(fpsAnimator)
    setSize(Size) 
    setLocationRelativeTo(null) 
    setVisible(true)
    fpsAnimator.start()
  }

  // Call initializer function 
  setup
}

File: java/src/main/java/com/celestia/csc155/MainFrame.java

package com.celestia.csc155;

import com.celestia.csc155.input.InputHandler;
import com.celestia.csc155.util.GLEventHandler;
import com.celestia.csc155.util.R;

import java.awt.*;
import java.awt.event.*;
import com.jogamp.opengl.GLEventListener;
import com.jogamp.opengl.awt.GLCanvas;
import com.jogamp.opengl.util.FPSAnimator;

import javax.swing.*;

public class MainFrame extends JFrame {

    private final double aspectRatio = R.util.aspect;
    private final int width = R.util.width;
    private final int height = R.util.height;
    private final int fps = R.util.fps;
    private final Dimension size = new Dimension(width, height);

    private final GLEventListener glEventHandler = new GLEventHandler();
    private final GLCanvas glCanvas = new GLCanvas();
    private final FPSAnimator fpsAnimator = new FPSAnimator(glCanvas, fps);

    public MainFrame() {
        setTitle(R.util.title);
        setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
        getContentPane().add(glCanvas);
        glCanvas.addGLEventListener(glEventHandler);
        glCanvas.addMouseListener((MouseListener) glEventHandler);
        glCanvas.addKeyListener((KeyListener) glEventHandler);
        glCanvas.addMouseWheelListener((MouseWheelListener) glEventHandler);
        glCanvas.setAnimator(fpsAnimator);
        setSize(size);
        setLocationRelativeTo(null);
        //setResizable(false);
        setVisible(true);
        fpsAnimator.start();
    }
}
